### PR TITLE
Create Budget App (Budgify)

### PR DESCRIPTION
**Pull request by alonsofl**
## Summary
Created a RoR app where users can sign up and add categories to group the transactions.

#### Details
- Added Devise gem to manage authentication.
- Added controllers for categories (groups) and entities.
- Added join table `entities_groups`.
- Added views for categories and entities, index, show, and new.
- Added models for User, Group, Entity, and EntitiesGroup.
- Added RoR linters.
- Added unit tests with RSpec and Capybara gems.

> Live Demo: [Link](https://pure-headland-81709.herokuapp.com/)
> You could use test user:
> Email: test@test.com
> Password: 123456

> [Video presentation](https://www.loom.com/share/d6125fc51d1540c6bd1b6ae729088e53)

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      @amounts[idx] = []
+      e.each do |ele|
+        if ele == 0
+          @amounts[idx].push(ele)
+        else
+          @amounts[idx].push(ele.amount)
+        end
+      end
+    end
+  end
```
Comment (henatan99) [inline app/controllers/categories_controller.rb:29]: - [ ] OPTIONAL: This indexing method for the category model can be written better. Try to use simpler more efficient methods like `includes`. You can also use `scope` method in the models instead.

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @alonsofl , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Davidosky007** [comment]
## Hi @alonsofl 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 
CR GIF



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alonsofl** [comment]
Thanks, @Davidosky007! :muscle:

**Pull request merged**